src/formats/fonts.c

```diff
@@ -57,18 +58,20 @@ int sd_font_save(const sd_font *font, const char *file) {
     return SD_SUCCESS;
 }
 
-int sd_font_decode(const sd_font *font, sd_vga_image *o, uint8_t ch, uint8_t color) {
+int sd_font_decode(const sd_font *font, sd_vga_image *o, uint8_t ch, uint8_t color, int transparent) {
     if(font == NULL || o == NULL || ch >= 224) {
         return SD_INVALID_INPUT;
     }
 
+    assert((transparent >= 0) && (transparent < 256));
+
     int t = 0;
     for(unsigned i = 0; i < font->h; i++) {
         for(int k = font->h - 1; k >= 0; k--) {
             if(font->chars[ch].data[i] & (1 << k)) {
                 o->data[t] = color;
             } else {
-                o->data[t] = 0;
+                o->data[t] = (char)transparent;
```

Comment:
shouldn't the transparent argument be a `uint8_t` like color? that would remove the need for the assert and the cast.

Reply:
hehe, just saw katajakasa's comment on some similar code below
unsure why it was marked as resolved, though

Reply:
Transparent can be -1 to indicate that the background is not transparent. For sd_font_decode that is not possible. Therefore the check is in place. To prevent this from breaking.